allensdk/brain_observatory/behavior/stimulus_processing.py

```diff
@@ -559,3 +559,61 @@ def is_change_event(stimulus_presentations: pd.DataFrame) -> pd.Series:
     is_change = is_change.fillna(False)
 
     return is_change
+
+
+def compute_trials_id_for_stimulus(
+        stim_pres_table: pd.DataFrame,
+        trials_table: pd.DataFrame) -> pd.Series:
+    """Add an id to allow for merging of the stimulus presentations
+    table with the trials table.
+
+    Parameters
+    ----------
+    stim_pres_table : pandas.DataFrame
+        Pandas stimulus table to create trials_id from.
+    trials_table : pandas.DataFrame
+        Trials table to create id from using trial start times.
+
+    Returns
+    -------
+    trials_id : pd.Series
+        Unique id to allow merging of the stim table with the trials table.
+        Null values are represented by -1.
+    """
+    trials_ids = pd.Series(
+        data=np.full(len(stim_pres_table), -1, dtype=int),
+        index=stim_pres_table.index,
+        name='trials_id')
+    trial_start_times = trials_table.start_time
+
+    stim_presentation_starts = stim_pres_table.start_time
+    stim_image_names = stim_pres_table.image_name
+
+    # Get the block ids for the behavior trial presentations
+    stim_blocks = stim_pres_table.stimulus_block
+    unique_stim_blocks = stim_pres_table[
+        ~stim_image_names.isna()].stimulus_block.unique()
+
+    # Get the first block, assume it is the 'active' block.
+    first_block_id = unique_stim_blocks[0]
```

Comment:
We can't assume this. For example if it starts with a gray screen as the first stim block.

Reply:
Check out line 595. This only looks for stimulus blocks that have images associated with them so it won't grab the grey screen block. I'll clarify the comment.